lib/cli/template/_LOCAL/wao-bind.js

```diff
@@ -0,0 +1,286 @@
+    }
+
+    // -------------------------------------
+    // ピリオドで分割した変数名のうち指定された順番の変数名を返す
+    // -------------------------------------
+    var getVarName = function(varPath, index) {
+      var varName = varPath.split(".");
+      if (varName.length < index) {
+        throw "Index Out Of Bounds. Variable Path's Count:=" + varName.length +
+          ", Index:=" + index + ", Variable name:=" + varPath;
+      }
+      return varName[index];
+    }
+
+    // -------------------------------------
+    // bind指定値を変数に変換する
+    // -------------------------------------
+    var getVar = function(varPath, org) {
+      var firstVarName = getVarName(varPath, 0);
+      var root = searchCore(firstVarName)[firstVarName];
+      if (root) {
+        var ret = root;
+        var paths = varPath.split(".");
+        var iteratorPath = firstVarName;
+        // 先頭はfirstVarName→rootで処理済みなので開始番号は1から
+        for (var i = 1; i < paths.length; i++) {
+          var trg = paths[i].split('|');
+          var wrapper;
+          if (trg.length > 1) {
+            trg = trg[0];
+            wrapper = getVar(varPath.replace(iteratorPath + '.' + trg +
+              '|', ''));
+          }
+          ret = getVarWithIterator(ret, iteratorPath)[trg];
+          if (wrapper) {
+            // TODO kimura これだとパイプを多重に指定できない・・・
+            ret = (typeof wrapper == 'function') ? wrapper(ret, org) :
+              wrapper[ret];
+            break;
+          }
+          iteratorPath += '.' + trg;
+        }
+      }
+      return ret;
+    }
+
+    // -------------------------------------
+    // iteratorがスタックされていればそれを返却する
+    // -------------------------------------
+    var getVarWithIterator = function(variable, iteratorPath) {
+      var retVar = variable;
+      var iterator = settings.iterators[iteratorPath];
+      if (iterator) {
+        retVar = iterator.this[iterator.idx];
+      } else if ($.isArray(retVar)) {
+        if (retVar.length == 1) {
+          retVar = retVar[0];
+        } else if (retVar.length == 0) {
+          throw "Bind data is not found.";
+        } else {
+          throw "Bind data is too many.";
+        }
+      }
+      return retVar;
+    }
+
+    // -------------------------------------
+    // data-wao-iteratorの実行
+    // -------------------------------------
+    var iterate = function($this, iteratorPath) {
+      // 予備動作
+      var $parent = $this.parent(); // 親要素を取得する
+      var $template = $this.removeAttr('data-wao-iterator').clone(); // 自分自身をテンプレート化
+      var $stack = $('<div />');
+      var iterator = getIterator(iteratorPath); // 繰り返し用の変数を取得しておく
+      $this.nextAll().each(function() {
+        // TODO kimura 削除対象にwao属性が残っていたらエラーにする予定
+        if ($(this).attr('data-wao') != 'keep' && !$(this).attr(
+            'data-wao-bind')) {
+          $(this).remove();
+        }
+      });
+      // 繰り返し処理本体
+      // iteratorは配列であることが保障されているのでfor-inを使用
+      if (iterator) {
+        for (var idx in iterator) {
```

Comment:
The body of a for in should be wrapped in an if statement to filter unwanted properties from the prototype.